### ISSUE TITLE
function that caculation the difference between two datapoints

### ISSUE THREAD

**Issue by seanlook** [crate/crate#7470]
I use cratedb for time series database, and some of my metrics is a counter value.

But when I draw the graph(like grafana), I want to show the **difference** (or **rate** concept in OpenTSDB, **increase** in promethus) rather than the total counter. There's no such function to do this as I look through the docs.

Or any advice?

**Comment by chaudum** [comment]
In SQL, this is usually achieved with the `LAG()` window function.
However, CrateDB does not support any window functions, yet, so deltas would need to be generated on the application side.

**Comment by mfussenegger** [comment]
We implemented window functions in 4.0 and with 4.1 we added some more window functions and framing option.

A delta or increment can be generated using `LAG(..)`, for example:

```sql
cr> SELECT 
      cnt, 
      cnt - lag(cnt, 1, 0) OVER (ORDER BY cnt) as delta 
    FROM 
      ( 
        VALUES 
          (1), 
          (47), 
          (53), 
          (70), 
          (72) 
        ) as t (cnt) ORDER BY 1 asc;                                                                                                                 
+-----+-------+
| cnt | delta |
+-----+-------+
|   1 |     1 |
|  47 |    46 |
|  53 |     6 |
|  70 |    17 |
|  72 |     2 |
+-----+-------+
```
